core/keychain/app_token.go

```diff
@@ -0,0 +1,75 @@
+package keychain
+
+import (
+	"errors"
+
+	"github.com/99designs/keyring"
+	"github.com/FleekHQ/space-daemon/core/permissions"
+)
+
+const AppTokenStoreKey = "appToken"
+const MasterAppTokenStoreKey = "masterAppToken"
+
+var ErrMasterTokenAlreadyExists = errors.New("master app token already exists")
+
+func (kc *keychain) StoreAppToken(tok *permissions.AppToken) error {
+	ring, err := kc.getKeyRing()
+	if err != nil {
+		return err
+	}
+
+	// Prevent overriding existing master key
+	key, _ := kc.st.Get([]byte(getMasterTokenStKey()))
+	if key != nil && tok.IsMaster {
+		return ErrMasterTokenAlreadyExists
+	}
+
+	// Prevents overriding even if user logged out and logged back in (which clears the store)
+	_, err = ring.Get(getMasterTokenStKey())
+	if err == nil && tok.IsMaster {
+		return ErrMasterTokenAlreadyExists
+	}
+
+	err = ring.Set(keyring.Item{
+		Key:   AppTokenStoreKey + "_" + tok.Key,
+		Data:  []byte(permissions.MarshalFullToken(tok)),
+		Label: "Space App - App Token",
```

Comment:
I don't really think it's necessary as this is just so that it gets displayed like that when you browse them in the keychain. It's not like the key to access the data later on.